scripts/dump-swagger.py

```diff
@@ -53,10 +53,39 @@ def resolve_references(path, schema):
     else:
         return schema
 
+def prefix_absolute_path_references(text, base_url):
+    """Adds base_url to absolute-path references.
+
+    Markdown links in descriptions may be absolute-path references.
+    These won’t work when the spec is not hosted at the root, such as
+    https://spec.matrix.org/latest/
+    This turns all `[foo](/bar)` found in text into
+    `[foo](https://spec.matrix.org/latest/bar)`, with
+    base_url = 'https://spec.matrix.org/latest/'
+    """
+    return text.replace("](/", "]({}/".format(base_url))
+
+def edit_links(node, base_url):
+    """Finds description nodes and makes any links in them absolute."""
+    if isinstance(node, dict):
+        for key in node:
+            if isinstance(node[key], str):
+                node[key] = prefix_absolute_path_references(node[key], base_url)
+            else:
+                edit_links(node[key], base_url)
+    elif isinstance(node, list):
+        for item in node:
+            edit_links(item, base_url)
 
 parser = argparse.ArgumentParser(
     "dump-swagger.py - assemble the Swagger specs into a single JSON file"
 )
+parser.add_argument(
+    "--base-url", "-b",
+    default="https://spec.matrix.org/latest/",
```

Comment:
we should use unstable as the default as it's what we end up assuming for the rest of the build:
```suggestion
    default="https://spec.matrix.org/unstable/",
```